ctrd/image_proxy_util.go

```diff
@@ -20,7 +20,11 @@ func SetImageProxy(p string) {
 }
 
 func proxyFromEnvironment(req *http.Request) (*url.URL, error) {
-	if proxy == "" || req.URL.Scheme == "https" {
```

Comment:
I want @yyb196 to help check this logic.

Reply:
if proxy is not blank, we should always use the proxy, no matter it can parse the http request headers or not, WDYT @Ace-Tang @zhuangqh . if proxy is not set, we use the result of `http.ProxyFromEnvironment`.

Reply:
The flag --image-proxy only acts on http request during pulling images. Using `http.ProxyFromEnvironment` by default for https request.  @Ace-Tang @yyb196

Reply:
@zhuangqh in semantic `--image-proxy` isn't limit to http proxy only, it's confuse to users who setup image proxy but the traffic isn't go through it, I this it is ok to use the image-proxy if schema of request is https. WDYT ?

Reply:
In my opinion, http and https proxy should be distinguished. Maybe we can split `--image-proxy` to `--image-http-proxy` and `--image-https-proxy`?